lib/data/repositories/user_repository.dart

```diff
@@ -0,0 +1,62 @@
+import 'dart:async';
+
+import 'package:memo/core/faults/errors/inconsistent_state_error.dart';
+import 'package:memo/data/gateways/sembast_database.dart';
+import 'package:memo/data/serializers/user_serializer.dart';
+import 'package:memo/domain/enums/memo_difficulty.dart';
+import 'package:memo/domain/models/user.dart';
+
+/// Handles all read, write and serialization operations pertaining to a [User]
+abstract class UserRepository {
+  /// Retrieves the current [User] and keeps listening to any changes made to it
+  Future<Stream<User>> listenToUser();
+
+  /// Retrieves the current [User]
+  Future<User> getUser();
+
+  /// Updates the [User] with the execution-related arguments
+  ///
+  /// Any update made to these properties, will override the current value, so make sure to update with the latest
+  /// corresponding values.
+  Future<void> updateExecution({required Map<MemoDifficulty, int> executionsAmounts, required int timeSpentInMillis});
+}
+
+final _userSerializer = UserSerializer();
+final _userStreamTransformer = StreamTransformer<Map<String, Object>, User>.fromHandlers(handleData: (rawUser, sink) {
+  sink.add(_userSerializer.from(rawUser));
+});
+
+class UserRepositoryImpl implements UserRepository {
+  UserRepositoryImpl(this._db);
+
+  final SembastDatabase _db;
+  final _mainStore = '';
```

Comment:
Why doesn't main store have a non-empty key? Like `main` or something else?

Reply:
That's a definition in sembast. Because how sembast access this main ref, I would have to create an specific case when accessing these refs in the main collection and, when I reference to a store with an empty value, it should default to the main collection (can't remember where I read about it now though)